Library/Homebrew/software_spec.rb

```diff
@@ -439,13 +439,30 @@ def checksums
       # Sort non-MacOS tags below MacOS tags.
       "0.#{tag}"
     end
-    sha256s = tags.reverse.map do |tag|
+    tags.reverse.map do |tag|
       {
-        collector[tag][:checksum] => tag,
-        cellar: collector[tag][:cellar],
+        "tag"    => tag,
+        "digest" => collector[tag][:checksum],
+        "cellar" => collector[tag][:cellar],
       }
     end
-    { sha256: sha256s }
+  end
+
+  def generate_sha256_line(tag, digest, cellar)
+    sha256_line = "sha256 \"#{digest}\" => :#{tag}"
+    if cellar.present?
+      default_cellars = [
+        Homebrew::DEFAULT_MACOS_CELLAR,
+        Homebrew::DEFAULT_MACOS_ARM_CELLAR,
+        Homebrew::DEFAULT_LINUX_CELLAR,
+      ]
+      if cellar.is_a? Symbol
+        sha256_line += ", :cellar => :#{cellar}"
+      elsif default_cellars.exclude?(cellar)
+        sha256_line += ", :cellar => \"#{cellar}\""
```

Comment:
Would be great, if possible, to have some tests that touch this line.

Reply:
Done